mongo/database.go

```diff
@@ -28,7 +28,7 @@ var (
 	defaultRunCmdOpts = []*options.RunCmdOptions{options.RunCmd().SetReadPreference(readpref.Primary())}
 )
 
-// Database performs operations on a given database.
+// Database is a handle to a MongoDB database. It is safe for concurrent use by multiple goroutines.
```

Comment:
I wasn't sure how much documentation we needed for the top-level `Database` type. I have this as a starting point, but am open to suggestions. I looked at pymongo docs for this and they don't have too much for this either (https://api.mongodb.com/python/current/api/pymongo/database.html#pymongo.database.Database.list_collection_names).